sgtypes/src/channel_transaction.rs

```diff
@@ -1,28 +1,196 @@
 // Copyright (c) The Starcoin Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use std::{
-    convert::TryFrom,
-    fmt::{Display, Formatter},
-};
-
-use serde::{Deserialize, Serialize};
-
+use crate::channel_transaction_sigs::ChannelTransactionSigs;
 use canonical_serialization::{
     CanonicalDeserialize, CanonicalDeserializer, CanonicalSerialize, CanonicalSerializer,
     SimpleDeserializer, SimpleSerializer,
 };
-use crypto::{
-    ed25519::{Ed25519PublicKey, Ed25519Signature},
-    HashValue,
-};
+use crypto::hash::{CryptoHash, CryptoHasher, TestOnlyHasher};
+use crypto::HashValue;
 use failure::prelude::*;
-use libra_types::transaction::TransactionArgument;
-use libra_types::{
-    account_address::AccountAddress,
-    transaction::{ChannelWriteSetBody, RawTransaction, TransactionOutput, Version},
-};
+use libra_types::transaction::{ChannelTransactionPayload, TransactionArgument, Version};
+use libra_types::{account_address::AccountAddress, transaction::TransactionOutput};
+use serde::{Deserialize, Serialize};
 use std::time::Duration;
+use std::{
+    convert::TryFrom,
+    fmt::{Display, Formatter},
+};
+
+/// sender (init channel transaction):
+/// 1. constructs ChannelTransaction, (sign on it if offchain)
+/// 2. from  ChannelTransaction, construct Libra RawTransaction, (sign on it of onchain),
+/// 3. execute it, get the channel writeset payload, and sign on it.
+/// 4. construct a SignedChannelTransaction, and send it to receiver.
+///
+/// receiver (verify channel transaction and sign on it):
+/// 1. check the signature on ChannelTransaction,
+/// 2. constructs the raw transaction, (mock sender signature on raw tx if offchain), execute it, get the writeset.
+/// 3. check the sender's signature on writeset payload.
+/// 4. sign on transaction and the writeset payload.
+/// 5. construct a SignedChannelTransaction, and send it to sender.
+///
+/// sender/reciever (apply channel tx):
+/// 1. check signature again if sender.
+/// 2. if onchian, sender constructs signed transaction of onchain, submit it to onchain.
+///    receiver waits the onchain tx.
+/// 3. if offchain, sender and receiver apply the tx to their local storage.  
+
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ChannelTransaction {
+    /// The global status version on this tx executed.
+    version: Version,
+    operator: ChannelOp,
+    /// txn sender
+    sender: AccountAddress,
+    /// Sequence number of this transaction corresponding to sender's account.
+    sequence_number: u64,
+    /// txn receiver
+    receiver: AccountAddress,
+    /// Sequence number of this channel.
+    channel_sequence_number: u64,
+    /// The txn expiration time
+    expiration_time: Duration,
+
+    args: Vec<TransactionArgument>,
+
+    /// Maximal total gas specified by wallet to spend for this transaction.
+    max_gas_amount: u64,
+    /// Maximal price can be paid per gas.
+    gas_unit_price: u64,
```

Comment:
max_gas_amount, gas_unit_price can be removed from ChannelTransaction. let The sender decide the gas value when submitting the transaction to the chain. 
The expiration_time can keep for verifying ChannelTransaction expiration but not is the RawTransaction 's expiration_time.

Reply:
Updated. gas related fileds are removed from channel transaction.

expiration_time is keeped for later usage.